internal/utils/flags/flags.go

```diff
@@ -6,97 +6,115 @@ import (
 	"fmt"
 	"sort"
 	"strings"
-
-	"github.com/spf13/pflag"
 )
 
 const (
 	//TypeString is the type of strings
 	TypeString = "string"
 )
 
-// enumSetValue is modified copy of stringSliceValue from
-// the cobra pflags package
-type enumSetValue struct {
-	value          *[]string
-	changed        bool
-	validValuesSet map[string]bool
+var setMember struct{}
+
+// EnumSetValue is modified copy of stringSliceValue from
+// the cobra pflags package.  It validates a set of enum values
+// and does not maintain order of the input arguments
+type EnumSetValue struct {
+	values         *[]string
+	valuesSet      map[string]struct{}
+	validValues    []string
+	validValuesSet map[string]struct{}
 }
 
-// NewEnumSetValue creates a stringSliceValue that validates inputs and prevents duplicates
-func NewEnumSetValue(p *[]string, val []string, validValues []string) pflag.Value {
-	esv := new(enumSetValue)
-	esv.value = p
-	*esv.value = val
-	esv.validValuesSet = make(map[string]bool)
+// NewEnumSet creates a EnumSetValue
+func NewEnumSet(p *[]string, defaultValues []string, validValues []string) *EnumSetValue {
+	esv := new(EnumSetValue)
+	esv.values = p
+	*esv.values = defaultValues
+	esv.validValues = validValues
+	esv.validValuesSet = make(map[string]struct{}, len(validValues))
 	for _, validValue := range validValues {
-		esv.validValuesSet[validValue] = true
+		esv.validValuesSet[validValue] = setMember
+	}
+	esv.valuesSet = make(map[string]struct{}, len(defaultValues))
+	for _, value := range defaultValues {
+		esv.valuesSet[value] = setMember
+	}
+	err := esv.validateAndRemoveDuplicates()
+	if err != nil {
+		// what should I do here?  I'm pretty sure this would be a great example of when
+		// to panic (coding error in a single threaded, non-critical application) but I know
+		// you don't like that so lmk what to do.
```

Comment:
this

Reply:
for this, i would either remove `defaultValues` all together (we don't/won't use them for array flags in this cli) and its corresponding validation in this constructor, or simply only include the defaults that are valid values, and silently ignore invalid ones here

Looking at this more, I also think this `EnumSetValue` would benefit from having a `func (es *EnumSetValue) set(vals ...string) error` function defined on it, so we can consolidate this validation/append logic and use them where need be in `Set`/`Append`/`Replace`.  you could use it here to, but just `set` with each individual default value so you could `continue` on any error